src/details/ArborX_DetailsHappyTreeFriends.hpp

```diff
@@ -45,7 +45,21 @@ struct HappyTreeFriends
   }
 
   template <class BVH>
+#if defined(__HIPCC__) && defined(__HIP_DEVICE_COMPILE__)
```

Comment:
Why is it not just `#ifdef __HIP_DEVICE_COMPILE__`?

Reply:
https://rocmdocs.amd.com/en/latest/Programming_Guides/HIP-porting-guide.html#compiler-defines-summary

My original code was
`#if defined(__HIP__) && defined(__HIP_DEVICE_COMPILE__)`
as I conservatively wanted to only enable it for the device code and HIP-Clang compiler. `__HIP__` was only defined for that.

@Rombur asked to change it to `__HIPCC__`. But that is now defined for both HIP-Clang and nvcc. So, I guess, now it's not necessary in the presence of `__HIP_DEVICE_COMPILE__`. I can remove `__HIPCC__` or reintroduce `__HIP__`.

Reply:
We are never compiling with `hipcc` to target NVIDIA hardware (not allowed by Kokkos) so this is pointless

Reply:
@Rombur please advise on how to best guard the code here

Reply:
Agree `__HIPCC__` is useless and so would be `__HIP__`